src/Microsoft.ML.AutoML/API/RankingExperiment.cs

```diff
@@ -26,10 +26,20 @@ public sealed class RankingExperimentSettings : ExperimentSettings
         /// The default value is a collection auto-populated with all possible trainers (all values of <see cref="RankingTrainer" />).
         /// </value>
         public ICollection<RankingTrainer> Trainers { get; }
+
+        /// <summary>
+        /// Maximum truncation level for computing (N)DCG
+        /// </summary>
+        /// <value>
+        /// The default value is 10.
+        /// </value>
+        public int DcgTruncationLevel { get; set; }
```

Comment:
We may want to call this parameter, `OptimizationDcgLevel`, since it will be the DCG/NDCG level which AutoML will sweep towards and use for selecting the best model.
```suggestion
        public int OptimizationDcgLevel { get; set; }
```

This is a conjoined twin to the `OptimizationMetric` parameter which, for ranking, has the choice of either DCG or NDCG. Then the user has to select _which_ DCG/NDCG, e.g. `OptimizationDcgLevel = 5` to select `NDCG@5` as the final metric the for optimization.

Then internally we can pass along a `DcgTruncationLevel = Math.Max(2 * OptimizationDcgLevel, 10)` to the evaluator, which ensures the user's requested optimization `k` is produced. 

The padding is incase the user is interested in the NDCG values with consideration for additional results (e.g. optimize towards NDCG@6 for an auto-suggest, but also tell me how good the first 10 results are).

Reply:
> Then internally we can pass along a DcgTruncationLevel = Math.Max(2 * OptimizationDcgLevel, 10) to the evaluator, which ensures the user's requested optimization k is produced.

If the user directly specifies the `OptimizationDcgLevel` do we need to multiply it by 2/ take the max compared to 10? Or with how AutoML works do we need to make sure it has extra than what they specify?

Reply:
No, the AutoML code doesn't rely on having the extra NDCG/DCG metrics. It specifically will look at `NDCG@k`/`DCG@k` where `k=OptimizationDcgLevel` when determining the winning model and measuring improvements during the sweeping process. The sample code for AutoML ranking ([PR](https://github.com/dotnet/machinelearning-samples/pull/852)) does display `NDCG@1,3,10` & `DCG@1`.

The larger DcgTruncationLevel is just for viewing purposes. The user may want want to optimize `NDCG@6` for an auto-suggest, but also want to know the `NDCG@10` metric.

Looking at the code, longer `@k` values (`truncationLevel`) are approximately free to calculate:
https://github.com/dotnet/machinelearning/blob/f87a3bbd8adb12934dae0a0060813ce9b7500664/src/Microsoft.ML.Data/Evaluators/RankingEvaluator.cs#L1043-L1050

Reply:
If this is the truncation level for both NDCG/DCG, will it be confusing that we name it Optimization**DCG**Level?

Reply:
@Lynx1820: It's a good question. 

The naming convention comes from the ranking evaluator's `DcgTruncationLevel` parameter:
https://github.com/dotnet/machinelearning/blob/f87a3bbd8adb12934dae0a0060813ce9b7500664/src/Microsoft.ML.Data/Evaluators/RankingEvaluator.cs#L37-L38

I'm open to other names. 

Maybe `OptimizationMetricTruncationLevel` to be more generic? Though users may misinterpret and assume "truncation" refers to us clipping to the metrics to N digits (e.g. 0.8479274 => 0.847).

One other aspect -- if we introduce additional ranking metrics, some will support this truncation limit and some will not. For instance `Precision@N` & `Recall@N`<sup>[[1](https://medium.com/swlh/rank-aware-recsys-evaluation-metrics-5191bba16832)]</sup>. I think this lends credence to the more generic form.

As the ol' adage goes:
> There are 2 hard problems in computer science: cache invalidation, naming things, and off-by-1 errors. -- Leon Bambrick

Reply:
@michaelgsharp: Is `OptimizationMetricTruncationLevel`, suitable? This represents the DCG/NDCG level which AutoML will sweep towards and use for selecting the best model, along with the truncation level for future metrics once we introduce them.
```suggestion
        public int OptimizationMetricTruncationLevel { get; set; }
```

Reply:
Changed and pushed.